src/patterns/menu/Menu.tsx

```diff
@@ -0,0 +1,133 @@
+                    {data.label && isMobile && (
+                        <>
+                            <Grid
+                                columnsTemplate={'auto 1fr'}
+                                placeItems={'center'}
+                                onClick={handleToggle}
+                            >
+                                <Icon glyph="chevron-left" />
+                                <MenuLabelRoot isMobile={isMobile}>{data.label}</MenuLabelRoot>
+                            </Grid>
+                            <Divider />
+                        </>
+                    )}
+                    {data.items?.map((item, index) => {
+                        const key = `${item.label}_${index}`;
+
+                        if (item.items && item.items.length > 0) {
+                            return <SubMenu key={key} data={item} isMobile={isMobile} />;
+                        }
+
+                        return (
+                            <MenuItem
+                                key={key}
+                                label={item.label}
+                                leadingElement={item.leadingElement}
+                                trailingElement={item.trailingElement}
+                            />
+                        );
+                    })}
+                </MenuRoot>
+            </Popover>
+        </>
+    );
+};
+
+const Menu = (props: PMenu): JSX.Element => {
+    const { title, trigger, isVisible, data, onClickAway, isMobile, placement, ...rest } = props;
+
+    return (
+        <Popover
+            anchorReference={trigger}
+            isVisible={isVisible}
+            onClickAway={onClickAway}
+            styles={isMobile ? mobileMenuStyles : undefined}
+            customTransition={isMobile ? MENU_TRANSITIONS_MAP.mainMenu : undefined}
+            placement={placement}
+            {...rest}
+        >
+            <MenuRoot isMobile={isMobile}>
+                {title && (
+                    <>
+                        <MenuLabelRoot isMobile={isMobile}>{title}</MenuLabelRoot>
+                        <Divider />
+                    </>
+                )}
+                {data.items?.map((item, index) => {
+                    const key = `${item.label}_${index}`;
+
+                    if (item.items && item.items.length > 0) {
+                        return <SubMenu key={key} data={item} isMobile={isMobile} />;
+                    }
+
+                    return (
+                        <>
+                            <MenuItem
+                                key={key}
+                                label={item.label}
+                                leadingElement={item.leadingElement}
+                                trailingElement={item.trailingElement}
+                            />
+                            {item.divider && <Divider />}
+                        </>
+                    );
+                })}
+            </MenuRoot>
```

Comment:
this feels like it could be just a `SubMenu` ... it might simplify things, but I will take a closer look at it later today to identify the possibilities here